File: original/modules/event_handler.py
# 事件处理模块
import re
import json
from typing import Dict, List, Callable, Any

class EventHandler:
    """事件处理类"""
    
    def __init__(self):
        self.event_handlers: Dict[str, List[Callable]] = {
            'join': [],
            'leave': [],
            'message': [],
            'dm': [],
            'music': [],
            'new_host': []
        }
        self.command_handlers: Dict[str, Callable] = {}
        self.regex_handlers: List[tuple] = []  # (pattern, handler)
        
    def register_event_handler(self, event_type: str, handler: Callable):
        """注册事件处理器"""
        if event_type in self.event_handlers:
            self.event_handlers[event_type].append(handler)
            
    def register_command_handler(self, command: str, handler: Callable):
        """注册命令处理器"""
        self.command_handlers[command.lower()] = handler
        
    def register_regex_handler(self, pattern: str, handler: Callable):
        """注册正则表达式处理器"""
        self.regex_handlers.append((re.compile(pattern), handler))
        
    async def handle_event(self, event_type: str, data: Dict[str, Any]):
        """处理事件"""
        # 处理事件处理器
        if event_type in self.event_handlers:
            for handler in self.event_handlers[event_type]:
                try:
                    await handler(data)
                except Exception as e:
                    print(f"事件处理器出错: {e}")
                    
    async def handle_message(self, message: str, user: Dict[str, Any]):
        """处理消息"""
        # 检查是否为命令
        if message.startswith('/'):
            await self.handle_command(message, user)
        else:
            # 检查正则表达式处理器
            for pattern, handler in self.regex_handlers:
                match = pattern.match(message)
                if match:
                    try:
                        await handler(message, user, match)
                    except Exception as e:
                        print(f"正则表达式处理器出错: {e}")
                        
    async def handle_command(self, command: str, user: Dict[str, Any]):
        """处理命令"""
        command_parts = command[1:].split()
        if not command_parts:
            return
            
        cmd = command_parts[0].lower()
        args = command_parts[1:] if len(command_parts) > 1 else []
        
        # 查找命令处理器
        if cmd in self.command_handlers:
            try:
                await self.command_handlers[cmd](cmd, args, user)
            except Exception as e:
                print(f"命令处理器出错: {e}")

File: app/src/main/java/com/drrr/bot/EventHandler.java
package com.drrr.bot;

import android.util.Log;
import java.util.List;
import java.util.ArrayList;
import java.util.Map;
import java.util.HashMap;

public class EventHandler {
    private static final String TAG = "EventHandler";
    
    // 事件处理器映射
    private Map<String, List<EventHandlerCallback>> eventHandlers;
    
    // 命令处理器映射
    private Map<String, EventHandlerCallback> commandHandlers;
    
    public EventHandler() {
        eventHandlers = new HashMap<>();
        commandHandlers = new HashMap<>();
        
        // 初始化事件类型
        eventHandlers.put("join", new ArrayList<EventHandlerCallback>());
        eventHandlers.put("leave", new ArrayList<EventHandlerCallback>());
        eventHandlers.put("message", new ArrayList<EventHandlerCallback>());
        eventHandlers.put("dm", new ArrayList<EventHandlerCallback>());
        eventHandlers.put("music", new ArrayList<EventHandlerCallback>());
        eventHandlers.put("new_host", new ArrayList<EventHandlerCallback>());
        
        Log.d(TAG, "EventHandler initialized");
    }
    
    /**
     * 注册事件处理器
     * @param eventType 事件类型
     * @param handler 处理器回调
     */
    public void registerEventHandler(String eventType, EventHandlerCallback handler) {
        if (eventHandlers.containsKey(eventType)) {
            eventHandlers.get(eventType).add(handler);
            Log.d(TAG, "Registered event handler for: " + eventType);
        } else {
            Log.w(TAG, "Unknown event type: " + eventType);
        }
    }
    
    /**
     * 注册命令处理器
     * @param command 命令
     * @param handler 处理器回调
     */
    public void registerCommandHandler(String command, EventHandlerCallback handler) {
        commandHandlers.put(command.toLowerCase(), handler);
        Log.d(TAG, "Registered command handler for: " + command);
    }
    
    /**
     * 处理事件
     * @param eventType 事件类型
     * @param data 事件数据
     */
    public void handleEvent(String eventType, Map<String, Object> data) {
        if (eventHandlers.containsKey(eventType)) {
            for (EventHandlerCallback handler : eventHandlers.get(eventType)) {
                try {
                    handler.onEvent(data);
                } catch (Exception e) {
                    Log.e(TAG, "Error in event handler for: " + eventType, e);
                }
            }
        }
    }
    
    /**
     * 处理消息
     * @param message 消息内容
     * @param user 用户信息
     */
    public void handleMessage(String message, Map<String, Object> user) {
        // 检查是否为命令
        if (message.startsWith("/")) {
            handleCommand(message, user);
        } else {
            // 处理普通消息事件
            Map<String, Object> data = new HashMap<>();
            data.put("message", message);
            data.put("user", user);
            handleEvent("message", data);
        }
    }
    
    /**
     * 处理命令
     * @param message 命令消息
     * @param user 用户信息
     */
    private void handleCommand(String message, Map<String, Object> user) {
        // 解析命令
        String[] parts = message.substring(1).split(" ", 2);
        String command = parts[0].toLowerCase();
        String args = parts.length > 1 ? parts[1] : "";
        
        // 查找命令处理器
        if (commandHandlers.containsKey(command)) {
            Map<String, Object> data = new HashMap<>();
            data.put("command", command);
            data.put("args", args);
            data.put("user", user);
            
            try {
                commandHandlers.get(command).onEvent(data);
            } catch (Exception e) {
                Log.e(TAG, "Error in command handler for: " + command, e);
            }
        } else {
            // 未找到命令处理器
            Log.d(TAG, "No handler found for command: " + command);
        }
    }
    
    /**
     * 事件处理器回调接口
     */
    public interface EventHandlerCallback {
        void onEvent(Map<String, Object> data);
    }
}